infra/base-images/base-runner/test_all.py

```diff
@@ -162,6 +172,46 @@ def test_all_outside_out(allowed_broken_targets_percentage):
     return test_all(out, allowed_broken_targets_percentage)
 
 
+def centipede_needs_auxiliaries():
+  """Checks if auxiliaries are needed for Centipede."""
+  # Centipede always requires unsanitized binaries as the main fuzz targets,
+  # and separate sanitized binaries as auxiliaries.
+  # 1. Building sanitized binaries with helper.py (i.e., local or GitHub CI):
+  # Unsanitized ones will be built automatically into the same docker container.
+  # Script bad_build_check tests both
+  # a) If main fuzz targets can run with the auxiliaries, and
+  # b) If the auxiliaries are built with the correct sanitizers.
+  # 2. In Trial build and production build:
+  # Two kinds of binaries will be in separated buckets / docker containers.
+  # Script bad_build_check tests either
+  # a) If the unsanitized binaries can run without the sanitized ones, or
+  # b) If the sanitized binaries are built with the correct sanitizers.
+  return (os.getenv('FUZZING_ENGINE') == 'centipede' and
+          os.getenv('SANITIZER') != 'none' and os.getenv('HELPER'))
+
+
+def find_centipede_auxiliary(main_fuzz_target_path):
+  """Finds the sanitized binary path that corresponds to |main_fuzz_target| for
+  bad_build_check."""
+  target_dir, target_name = os.path.split(main_fuzz_target_path)
+  sanitized_binary_dir = os.path.join(target_dir,
+                                      f'__centipede_{os.getenv("SANITIZER")}')
```

Comment:
single quotes.

Reply:
I am not sure if I understand it correctly, but single quotes around `SANITIZER` will break the single quote of `f-sring`:
```python
  File "/usr/local/bin/test_all.py", line 198
    f'__centipede_{os.getenv('SANITIZER')}')
                              ^
SyntaxError: invalid syntax
```
Escaping single quotes is not allowed in `f-string` either:
```python
  File "/usr/local/bin/test_all.py", line 198
    f'__centipede_{os.getenv(\'SANITIZER\')}')
    ^
SyntaxError: f-string expression part cannot include a backslash
```

Reply:
Oh nevermind. You were right initially and I was wrong.